src/lotus/mod.rs

```diff
@@ -0,0 +1,41 @@
+pub mod client;
+///! The lotus api to interact with lotus node
+pub mod types;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use cid::Cid;
+use fvm_shared::address::Address;
+use serde::de::DeserializeOwned;
+use std::fmt::Debug;
+
+pub use crate::lotus::client::LotusClient;
+pub use crate::lotus::types::{MpoolPushMessage, MpoolPushMessageInner};
+pub use crate::lotus::types::{
+    ReadStateResponse, StateWaitMsgResponse, WalletKeyType, WalletListResponse,
+};
+
+#[async_trait]
+pub trait LotusApi {
```

Comment:
There is no need to change name. I assume even if it's http/ws, it's still the same interface.

Reply:
I'm sorry, but I'm going to have to insist that you rename this. It might not sound like a big deal, but this sort of thing adds up to the cognitive load required to understand the codebase. For instance, Lotus is riddled with types named in such a way (e.g., NetApi, FullNodeApi, WalletApi, etc.), making much of the code really difficult to read because of the lack of specificity in the naming. This trait is an interface for a JSON-RPC client that sends requests to a Lotus node. Please call it what it is, a `LotusJsonRpcClient`. By calling it `LotusApi`, with don't know if it's a client, a server, or something entirely different. We really don't know what to expect when seeing an object of such a type in the code since the name is just too generic.